src/VisualStudio/Core/Impl/SolutionExplorer/AnalyzersFolderItem/AnalyzersFolderItemSource.cs

```diff
@@ -4,48 +4,30 @@
 
 using System.Collections;
 using System.Collections.ObjectModel;
-using System.Linq;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.VisualStudio.Shell;
 
 namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer;
 
-internal sealed class AnalyzersFolderItemSource : IAttachedCollectionSource
+internal sealed class AnalyzersFolderItemSource(
+    IThreadingContext threadingContext,
+    Workspace workspace,
+    ProjectId projectId,
+    IVsHierarchyItem projectHierarchyItem,
+    IAnalyzersCommandHandler commandHandler)
+    : IAttachedCollectionSource
 {
-    private readonly IVsHierarchyItem _projectHierarchyItem;
-    private readonly Workspace _workspace;
-    private readonly ProjectId _projectId;
-    private readonly ObservableCollection<AnalyzersFolderItem> _folderItems;
-    private readonly IAnalyzersCommandHandler _commandHandler;
-
-    public AnalyzersFolderItemSource(Workspace workspace, ProjectId projectId, IVsHierarchyItem projectHierarchyItem, IAnalyzersCommandHandler commandHandler)
-    {
-        _workspace = workspace;
-        _projectId = projectId;
-        _projectHierarchyItem = projectHierarchyItem;
-        _commandHandler = commandHandler;
-
-        _folderItems = [];
-
-        Update();
-    }
+    private readonly ObservableCollection<AnalyzersFolderItem> _folderItems = [new AnalyzersFolderItem(
+        threadingContext,
+        workspace,
+        projectId,
+        projectHierarchyItem,
+        commandHandler.AnalyzerFolderContextMenuController)];
 
     public bool HasItems => true;
 
     public IEnumerable Items => _folderItems;
 
-    public object SourceItem => _projectHierarchyItem;
-
-    internal void Update()
```

Comment:
this was only called in the constructor.  so it could be inlined.  once inlined, it meant that almost all state held int his type went away.